misc/git-hooks/pre-commit

```diff
@@ -33,6 +33,13 @@ fi
 export git_ls_files
 git_ls_files="$(git ls-files)"
 
+function check_forbiden_depenencies() {
+  if grep -E "(fedimint-mint|fedimint-wallet|fedimint-ln)" fedimint-server/Cargo.toml >&2 ; then
+    >&2 echo "fedimint-server/Cargo.toml must not depend on modules"
+    return 1
+  fi
+}
+export -f check_forbiden_depenencies
```

Comment:
Can we do the same for others like new client <-> server?

Reply:
Which crates shouldn't include which exactly?

Reply:
Citing recent example, `gateway/*` should not include `fedimint-server`. #1723